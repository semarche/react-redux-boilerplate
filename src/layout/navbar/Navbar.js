import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import './styles.scss'

const propTypes = {
  authentication: PropTypes.func.isRequired,
};

class Navbar extends Component {
  render() {
    const { authentication } = this.props;
    return (
      <div className={'top-navbar'}>
        <div className={'left-part'}>
          <NavLink
            to={'/'}
            activeClassName="selected"
          >Home</NavLink>
          <NavLink
            to={'/profile'}
            activeClassName="selected"
          >Profiles</NavLink>
        </div>
        <div className={'right-part'}>
          <h4 onClick={authentication}>Log out</h4>
        </div>
      </div>
    )
  }
}

Navbar.propTypes = propTypes;
export default Navbar;
