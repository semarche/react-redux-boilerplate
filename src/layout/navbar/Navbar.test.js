import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Navbar  from './Navbar';

configure({ adapter: new Adapter() });

describe('Navbar', () => {
  const wrapper = shallow(
    <Navbar
      authentication={jest.fn()}
    />
  );
  it('should render top-navbar', () => {
    expect(wrapper.find('.top-navbar').exists()).toEqual(true);
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
