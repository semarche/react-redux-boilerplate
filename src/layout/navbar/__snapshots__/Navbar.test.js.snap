// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches the snapshot 1`] = `
<div
  className="top-navbar"
>
  <div
    className="left-part"
  >
    <NavLink
      activeClassName="selected"
      aria-current="page"
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="selected"
      aria-current="page"
      to="/profile"
    >
      Profiles
    </NavLink>
  </div>
  <div
    className="right-part"
  >
    <h4
      onClick={[MockFunction]}
    >
      Log out
    </h4>
  </div>
</div>
`;
